README.md

```diff
@@ -99,27 +99,34 @@ name the -B and -T options allow a mapping file to be specified to
 rename branches and tags (respectively). The syntax of the mapping
 file is the same as for the author mapping.
 
-Content filtering
+Plugins
 -----------------
 
-hg-fast-export supports filtering the content of exported files.
-The filter is supplied to the --filter-contents option. hg-fast-export
-runs the filter for each exported file, pipes its content to the filter's
-standard input, and uses the filter's standard output in place
-of the file's original content. The prototypical use of this feature
-is to convert line endings in text files from CRLF to git's preferred LF:
+hg-fast-export supports plugins to manipulate the file data and commit
+metadata. The plugins are enabled with the --plugin option. The value
+of said option is a plugin name (by folder in the plugins directory),
+and optionally, and equals-sign followed by an initialization string. 
+
+There is a readme accompanying each of the bundled plugins, with a 
+description of their usage. To create a new plugin, one must simply
+add a new folder under the `plugins` directory, with the name of their
+new plugin. Inside, there must be an `__init__.py` file, which contains
+at a minimum:
 
 ```
--- Start of crlf-filter.sh --
-#!/bin/sh
-# $1 = pathname of exported file relative to the root of the repo
-# $2 = Mercurial's hash of the file
-# $3 = "1" if Mercurial reports the file as binary, otherwise "0"
-
-if [ "$3" == "1" ]; then cat; else dos2unix; fi
--- End of crlf-filter.sh --
+def build_filter(args):
+    return Filter(args)
+
+class Filter:
+    def __init__(self, args):
+        pass
+        #Or don't pass, if you want to do some init code here
 ```
 
+Beyond the boilerplate initialization, you can see the two different 
+defined filter methods in the [dos2unix](./plugins/dos2unix) and 
+[branch_name_in_commit](./plugins/branch_name_in_commit) plugins.
+
```

Comment:
Please expand on the filter methods, i.e. explain what they are used for, their signature and what they expect in their input dict.